### PR TITLE
Implement task discussion

### PR DESCRIPTION
**Pull request by McSinyx**
~Somehow there isn't a ticket for this and~ I'm too lazy ~to create one~.  Resolve GH-56.

### CODE REVIEW

**Comment by Huy-Ngo** [comment]
> Somehow there isn't a ticket for this

Actually there is #56

**Comment by Huy-Ngo** [comment]
Hmm actually it's a composite one that hasn't been decomposed yet.

**Comment by McSinyx** [comment]
Fix me if I'm wrong, but I think this patch does all of it.

File: src/acanban/templates/task.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{# Task view page
+ # Copyright (C) 2020  Ngô Ngọc Đức Huy
```
Comment (Huy-Ngo) [inline src/acanban/templates/task.html:2]: I don't remember writing this actually.

Comment (McSinyx) [reply src/acanban/templates/task.html:2]: You don't?  You memory must be failing then!  I strongly recommend using ECC in your case.

**Comment by Huy-Ngo** [comment]
Is there a way I can test its UI?

**Comment by McSinyx** [comment]
Yes, sirry, please see the tests for a route.

File: tests/assets/rethinkdump/test/projects.json [hunk 2]
```diff
@@ -79,6 +81,92 @@
...
+                "created_on": {
+                  "$reql_type$": "TIME",
+                  "epoch_time": 1609023842,
+                  "timezone": "+07:00"
+                },
+                "replies": [
+                  {
+                    "creator": "arabellaf",
+                    "content": "Wow this is really\nhelpful, thanks.",
+                    "created_on": {
```
Comment (Huy-Ngo) [inline tests/assets/rethinkdump/test/projects.json:127]: I added these newlines for the diagram generation. It doesn't make sense for these to be in actual data.

File: src/acanban/templates/task.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+{% macro render_comment(comment, istask=false) -%}
+<div> <!-- forms are selected as adjacent to checkboxes to toggle -->
+  {{ comment.creator|userlink|safe }}
+  &bull;
+  {{ comment.created_on|naturaltime|safe }}
+  &bull;
+  {% set id = comment.created_on.timestamp() %}
+  <label for=toggle-{{ id }} class=reply-label>reply</label>
+  <input type=checkbox class=reply-toggle id=toggle-{{ id }}><br>
```
Comment (Huy-Ngo) [inline src/acanban/templates/task.html:28]: Why is this checkbox?

**Review by Huy-Ngo — CHANGES_REQUESTED** [review]
-1 from the UI/UX viewpoint.

- Displaying a reply box for every comment makes it very disruptive and space-consuming. It should only appear when the user want to reply to a comment. (I suppose you intentionally do this to avoid JavaScript?)
- I don't get what the checkboxes are for. Checked or unchecked, they seem to have no effects
- The comment forest is not intuitively illustrated. This is also what twitter and mastodon is lacking. Reddit and Lemmy seem to do this well.

File: tests/assets/rethinkdump/test/projects.json (hunk 2 above)
Comment (McSinyx) [reply tests/assets/rethinkdump/test/projects.json:127]: No, but it'll be a nice example of real life user input so I decided to keep them.  Probably it'll be useful once we render them as markdown (where single newlines are also no-op).

File: src/acanban/templates/task.html (hunk 3 above)
Comment (McSinyx) [reply src/acanban/templates/task.html:28]: Reload the CSS :wink:

Comment (Huy-Ngo) [reply src/acanban/templates/task.html:28]: Lol ok my bad (or web bad :thinking: ?).

Comment (McSinyx) [reply src/acanban/templates/task.html:28]: Soft-reloading is the null-able pointer of the web :shrug:

**Review by Huy-Ngo — APPROVED** [review]
Why doesn't CSS just reload :upside_down_face:

**Comment by McSinyx** [comment]
Agile, here I come!

**Pull request merged**